katana/wui/core/static/core/js/main.js

```diff
@@ -836,6 +836,150 @@ var katana = {
 		},
 
 	},
+
+	openFileExplorer(heading, callBack_on_accept, callBack_on_dismiss){
+	    if(!heading || heading == "" || heading == undefined){
+	        heading = "Select a file"
+	    }
+
+	    if(callBack_on_accept == undefined){
+	        callBack_on_accept = false;
+	    }
+
+	    if(callBack_on_dismiss == undefined){
+	        callBack_on_dismiss = false;
+	    }
+
+	    $.ajax({
+                headers: {
+                    'X-CSRFToken': wapp_management.getCookie('csrftoken')
+                },
+                type: 'GET',
+                url: 'get_file_explorer_data/',
+            }).done(function(data) {
+                var explorer_modal_html = '<div class="overlay">' +
```

Comment:
this html should be in an <script type="text/template" id="example"></script> somewhere in the html page and the js should recover it by doing something like $($('#example').html()). please do not put large html blocks into js.

Reply:
Addressed